### PR TITLE
R127

### PR DESCRIPTION
**Pull request by chuwy**
The release:

* Adds two new mobile schemas
* Makes `com.snowplowanalytics.oss/oss_context/jsonschema/1-0-0` public
* Adds `com.snowplowanalytics.oss/oss_context/jsonschema/1-0-1`
* Patches `com.snowplowanalytics.monitoring.batch/alert/jsonschema/1-0-0` - the patch is safe because schema could not be used with warehouses

### CODE REVIEW

**Review by istreeter — APPROVED** [review]

**Comment by istreeter** [comment]
This is not critical, but would be nice to also bump igluctl to 0.8.0 in the github action.

**Review by paulboocock — APPROVED** [review]

**Comment by chuwy** [comment]
> This is not critical, but would be nice to also bump igluctl to 0.8.0 in the github action.

Discussed this with Ian and decided to do another igluctl-only release later.

File: schemas/com.snowplowanalytics.mobile/deep_link/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"type": "object",
+	"properties": {
+		"url": {
+			"type": "string",
+			"description": "URL in the received deep-link",
+			"format": "uri",
+			"maxLength": 4096
+		},
+		"referrer": {
+			"type": "string",
```
Comment (jbeemster) [inline schemas/com.snowplowanalytics.mobile/deep_link/jsonschema/1-0-0:19]: This property is not required - should it be nullable in that case?

File: schemas/com.snowplowanalytics.mobile/application_lifecycle/jsonschema/1-0-0 [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  },
+
+  "type": "object",
+  "properties": {
+    "isVisible": {
+      "description": "Indicates if the app is in foreground state (true) or background state (false)",
+      "type": "boolean"
+    },
+    "index": {
+      "description": "Represents the foreground index or background index (tracked with com.snowplowanalytics.snowplow application_foreground and application_background events.",
```
Comment (jbeemster) [inline schemas/com.snowplowanalytics.mobile/application_lifecycle/jsonschema/1-0-0:18]: This property is not required - should it be nullable?

File: schemas/com.snowplowanalytics.mobile/deep_link_received/jsonschema/1-0-0 [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	},
+	"type": "object",
+	"properties": {
+		"url": {
+			"type": "string",
+			"description": "URL in the received deep-link",
+			"format": "uri",
+			"maxLength": 4096
+		},
+		"referrer": {
```
Comment (jbeemster) [inline schemas/com.snowplowanalytics.mobile/deep_link_received/jsonschema/1-0-0:18]: This property is not required - should it be nullable?

File: schemas/com.snowplowanalytics.monitoring.batch/alert/jsonschema/1-0-0 [hunk 4]
```diff
@@ -36,10 +36,7 @@
     },
     "tags": {
       "description": "Set of key value pairs providing additional information",
-      "type": "object",
-      "additionalProperties": {
-        "type": "string"
-      }
+      "type": "object"
```
Comment (jbeemster) [inline schemas/com.snowplowanalytics.monitoring.batch/alert/jsonschema/1-0-0:39]: @chuwy where is this schema used?  Assuming this cannot break any load processes / schema evolution through this removal?

File: schemas/com.snowplowanalytics.oss/oss_context/jsonschema/1-0-0 [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Context schema for OSS terraform modules, VM instances and application events",
+  "self": {
+    "vendor": "com.snowplowanalytics.oss",
+    "name": "oss_context",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
```
Comment (jbeemster) [inline schemas/com.snowplowanalytics.oss/oss_context/jsonschema/1-0-0:11]: This schema is not setting any required properties - seems to be something we do on all other schemas?

File: schemas/com.snowplowanalytics.oss/oss_context/jsonschema/1-0-1 [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Context schema for OSS terraform modules, VM instances and application events",
+  "self": {
+    "vendor": "com.snowplowanalytics.oss",
+    "name": "oss_context",
+    "format": "jsonschema",
+    "version": "1-0-1"
+  },
+  "type": "object",
+  "properties": {
```
Comment (jbeemster) [inline schemas/com.snowplowanalytics.oss/oss_context/jsonschema/1-0-1:11]: This schema is not setting any required properties - seems to be something we do on all other schemas?

File: schemas/com.snowplowanalytics.monitoring.batch/alert/jsonschema/1-0-0 (hunk 4 above)
Comment (jbeemster) [reply schemas/com.snowplowanalytics.monitoring.batch/alert/jsonschema/1-0-0:39]: Saw the comment on the PR - no issues.

**Comment by chuwy** [comment]
Hey @jbeemster,

Regarding all your comments about `required` and/or `null`. For downstream consumers (Loaders) there's no difference between: `{"foo": 42, "bar": null}` and `{"foo": 42}`. In both cases `bar` will be a nullable column.

For schemas there are two ways though to express optionality and it mostly depends on the producers themselves. The rule of thumb that I use to decide between "explicit null" and "missing key" is:

1. If it's a context sent over the network - use "missing key" approach, it can save some bites
2. If it's a some kind of configuration schema - use "explicit key", it will make templating easier and increase readability

Otherwise - rely on producer's preference.

**Pull request merged**